src/ethereum_clis/clis/execution_specs.py

```diff
@@ -132,39 +132,33 @@ def _generate_post_args(
 class ExecutionSpecsExceptionMapper(ExceptionMapper):
     """Translate between EEST exceptions and error strings returned by ExecutionSpecs."""
 
-    reliable: ClassVar[bool] = False
-    """
-    TODO: Exception messages returned from ExecutionSpecs are not reliable because most of the
-    exceptions consist of the same string without indication of the particular exception type.
-    """
-
     mapping_substring: ClassVar[Dict[ExceptionBase, str]] = {
-        TransactionException.TYPE_4_EMPTY_AUTHORIZATION_LIST: "Failed transaction: InvalidBlock()",
+        TransactionException.TYPE_4_EMPTY_AUTHORIZATION_LIST: "Failed transaction: EmptyAuthorizationListError('empty authorization list')",
         TransactionException.SENDER_NOT_EOA: "Failed transaction: InvalidSenderError('not EOA')",
-        TransactionException.TYPE_4_TX_CONTRACT_CREATION: "Failed transaction: ",
-        TransactionException.INSUFFICIENT_ACCOUNT_FUNDS: "ailed transaction: ",
-        TransactionException.TYPE_3_TX_MAX_BLOB_GAS_ALLOWANCE_EXCEEDED: "iled transaction: ",
-        TransactionException.INSUFFICIENT_MAX_FEE_PER_BLOB_GAS: "led transaction: ",
-        TransactionException.INSUFFICIENT_MAX_FEE_PER_GAS: "ed transaction: ",
+        TransactionException.TYPE_4_TX_CONTRACT_CREATION: "Failed transaction: TransactionTypeContractCreationError('transaction type `SetCodeTransaction` not allowed to create contracts')",
```

Comment:
Could we update those that have the precise exception type to just just expect that like this?
```suggestion
        TransactionException.TYPE_4_TX_CONTRACT_CREATION: "TransactionTypeContractCreationError",
```
It'll keep things cleaner :)

Reply:
sure! I have updated to use only the execution type wherever it was specific.

some thoughts triggered during mutation testing though - is that some mutations involved changing error msgs. With this change those mutations would not be detected. So I was wondering wether a stricter comparison would add any value at all here? :)